projects/feature-activation/0005-feature-activation-for-transactions.md

```diff
@@ -0,0 +1,196 @@
+
+After exploring the previous ideas, I determined that a _theoretical_ great solution would be to add a block parent to all transactions. The state of a transactions would be the state of its parent block. That would mean we wouldn't have to deal with reorgs, which is the main source of difficulties, since voiding a parent block would automatically void its transaction dependencies. However, in practice, making this change would be too risky as it could affect multiple dynamics of the network.
+
+Then, we arrived at the solution proposed in this document, which has most of the same advantages from the parent block solution, however it does not require any breaking change whatsoever. Here it is: **for a feature, a transaction is considered `ACTIVE` if its closest block, that is, its ancestor block with the greatest height, is `ACTIVE`**. This will be detailed in the Reference-level section.
+
+In other words, transactions today can already point to blocks via their inputs. Only a small portion of transactions spend from blocks, though. The idea is then to propagate this indirect block dependency to all transactions. This can be achieved by simply creating a new metadata that is set when a transaction is received from the network. The calculation is deterministic, equal for all peers, and with constant execution complexity. Also, reorgs are handled automatically by existing code, and no change has to be made in critical components such as the consensus.
+
+Equipped with this definition, the solution is complete. No other edge cases have to be handled. Also, to be clear, no change in the Feature Activation for Blocks is necessary. There are some use case limitations, though, explained in the [drawbacks] section.
+
+# Reference-level explanation
+[Reference-level explanation]: #reference-level-explanation
+
+In this section, technical details are expanded for what was described above. Before detailing the solution, let's formalize the context.
+
+## Overview
+
+### Premises
+
+1. After an `ACTIVE` block, all next blocks will also be `ACTIVE` (considering the same blockchain).
+2. Feature states for transactions are only `ACTIVE` or `INACTIVE`.
+
+### Requirements
+
+Using the premises above, we must define a function that returns a state, given a transaction (note: we actually return multiple states for multiple features). That function must satisfy the following requirements:
+
+1. A while after one `ACTIVE` block in the best chain, it must be possible to create `ACTIVE` transactions.
+2. A transaction that has an `ACTIVE` transaction as a dependency, is also `ACTIVE`.
+
+### Definition
+
+Repeated here from the Guide-level section:
+
+**For a feature, a transaction is considered `ACTIVE` if its closest block, that is, its ancestor block with the greatest height, is `ACTIVE`.**
```

Comment:
I guess we should be careful with the "closest block" expression since it might be misleading. Maybe use the expression `closest ancestor block` or `closest pointed block`.  I also think that you should expand the explanation to clarify that the transaction itself might be spending a mining reward which will make this block a candidate. I believe that two or three examples should clarify all cases.

Reply:
I think the proposed mechanism is very good, especially for its simplicity, but I also think "closest block" is not a great expression. I'll think of some alternatives and suggest here.

Reply:
This was renamed to `closest_ancestor_block` in the implementation, and I just updated it here.